api/controllers/models.py

```diff
@@ -416,3 +419,74 @@ def upload_to_s3(credentials):
             {"model_id": model.id, "s3_uri": f"s3://{bucket_name}/{s3_path}"}
         )
     )
+
+
+@bottle.get("/models/<mid:int>/deploy")
+@_auth.requires_auth
+def deploy_model_from_s3(credentials, mid):
+    # Authentication (only authenticated users can redeploy models for interaction)
+    u = UserModel()
+    user_id = credentials["id"]
+    user = u.get(user_id)
+    if not user:
+        logger.error("Invalid user detail for id (%s)" % (user_id))
+        bottle.abort(404, "User information not found")
+
+    m = ModelModel()
+    model = m.getUnpublishedModelByMid(mid)
+
+    model_owner = model.uid == user.id
+
+    if (not model.is_published) and (not model_owner):
+        bottle.abort(403, "Model is not published and user is not model owner")
```

Comment:
meaning, the only times a user should be able to `Request redeployment` is when they own the model, or the model is published